### DISCUSSION TITLE
Disable de-duplication?

### DISCUSSION THREAD

**Discussion by reinhard-brandstaedter** [icloud-photos-downloader/icloud_photos_downloader#875] [Q&A] [answered]
I'm having issues with the deduplication strategy. By default it adds a number (file-size?) to the name. However I want it to not do that and only identify by file name. Can deduplication be set to do that?

My usecase is as follows:
I'm downloading files to a local (temporary) folder. Then I have another job that goes through that folder and copies those images to the final location (doing some processing, adding metadata, etc.) in my library. This job then truncates the temporary files to size 0 but leaves the file as it has been downloaded by icloudpd.
When icloudpd runs the next time it thinks there are duplicates becasue it not only compares the filename but also recognizes the size difference and creates another file (with extension).
I want to not do that, and completely turn deduplication off or make it only name based.

Is that possible?

**Comment by AndreyNikiforov** [comment]
Yes, your request to support turning off de-duplication completely sounds reasonable. 

Since 1.20.0 you can [use file name with id](https://icloud-photos-downloader.github.io/icloud_photos_downloader/naming.html#duplicates) and it will not create extra files. It is not exactly what you need, but a half way in that direction.

**Reply by reinhard-brandstaedter** [reply]
I saw this in the documentation, but it's a bit unclear what it actually does.

Would this always create the same filename based on the source file?
--file-match-policy name-id7

**Reply by AndreyNikiforov** [reply, marked as answer]
> I saw this in the documentation, but it's a bit unclear what it actually does.

If you see a way to improve doc - pls share or even submit PR.

> --file-match-policy name-id7

images will be named as "IMG_1234_QAZXSW.JPG" instead of "IMG_1234.JPG", where QAZXSW is unique ID assigned by Apple to each asset (asset > image, e.g. live photo asset will have image and video). There will be no attempt to deduplicate - if there is a file with that name (regardless of the size), new download will be skipped.

**Reply by reinhard-brandstaedter** [reply]
Super, thanks!
That explains it really well and will exactly fit the usecase! I wasn't aware that this ID is assigned by apple already!
